test/cql-pytest/test_allow_filtering.py

```diff
@@ -291,3 +291,23 @@ def test_contains_frozen_collection_ck(cql, test_keyspace):
             "SELECT * FROM " + table + " WHERE a=0 AND c=0 AND b CONTAINS 0 ALLOW FILTERING")))
         assert 1 == len(list(cql.execute(
             "SELECT * FROM " + table + " WHERE a=0 AND c=0 AND b CONTAINS KEY 0 ALLOW FILTERING")))
+
+# table4 contains example table from issue #8991
+@pytest.fixture(scope="module")
+def table4(cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    cql.execute(f"CREATE TABLE {table} (k int, c1 int, c2 int, PRIMARY KEY (k,c1,c2))")
+    cql.execute(f"CREATE INDEX ON {table} (c2)")
+    for c1 in [0, 1, 2]:
+        for c2 in [0, 1, 2]:
+            cql.execute(f"INSERT INTO {table} (k, c1, c2) VALUES (0, {c1}, {c2})")
+    everything = list(cql.execute(f"SELECT * FROM {table}"))
+    wait_for_index(cql, table, 'c2', everything)
+    yield (table, everything)
+    cql.execute(f"DROP TABLE {table}")
+
+# Reproduces example from issue #8991
+# Selecting from indexed table should using only clustering key should require filtering
+@pytest.mark.xfail(reason="Select from indexed table using only clustering key should require filtering. Issue #8991")
```

Comment:
Honestly, I'd refer #7608 here.  I really really want to separate all af_mandatory considerations to that issue.

Reply:
I suppose it makes a difference whether an index is present or not, but even that seems like a minor variation of #7608.

Reply:
It might seem strange to someone that there is a separate test from different from the one already marked #7608
Maybe I could mark it as `a variation of #7608 found in #8991`

Reply:
> It might seem strange to someone that there is a separate test from different from the one already marked #7608

If so, that someone is already feeling strange.  We already have multiple testcases referencing the same issue (eg, 8717).

Reply:
It is perfectly fine to have multiple test cases for the same issues, if there's something interesting or different between them. In this case, the new test case has the distinction that it both incorrectly allows  ALLOW FILTERING to be dropped, and produces wrong results in this case, so it is interesting, because conceivable even if we required ALLOW FILTERING, the results with ALLOW FILTERING could still be wrong (and this is indeed what happened).
So I like your new test cases.
As long as we make sure that additional test cases are clearly documented, refer to specific issues, and take milliseconds to run and not seconds or minutes - then we don't need to be cheap on tests.